src/Microsoft.AspNetCore.Mvc.Api.Analyzers/SymbolApiResponseMetadataProvider.cs

```diff
@@ -276,16 +276,150 @@ internal static int GetStatusCode(AttributeData attribute)
 
                 return new ActualApiResponseMetadata(returnStatementSyntax, statusCode.Value);
             }
-            else if (!symbolCache.IActionResult.IsAssignableFrom(statementReturnType))
+
+            if (!symbolCache.IActionResult.IsAssignableFrom(statementReturnType))
             {
                 // Return expression does not have a DefaultStatusCodeAttribute and it is not
                 // an instance of IActionResult. Must be returning the "model".
                 return new ActualApiResponseMetadata(returnStatementSyntax);
             }
 
+            if (returnExpression is InvocationExpressionSyntax invocation)
+            {
+                // Covers the 'return StatusCode(200)' case.
+                if (TryGetParameterStatusCode(semanticModel, invocation.Expression, invocation.ArgumentList, cancellationToken, out var statusCode))
+                {
+                    return new ActualApiResponseMetadata(returnStatementSyntax, statusCode);
+                }
+
+                return null;
+            }
+
+            if (returnExpression is ObjectCreationExpressionSyntax creation)
+            {
+                // Covers the 'return new ObjectResult(...) { StatusCode = 200 }' case.
+                if (TryGetInitializerStatusCode(semanticModel, creation.Initializer, cancellationToken, out var statusCode))
+                {
+                    return new ActualApiResponseMetadata(returnStatementSyntax, statusCode);
+                }
+
+                // Covers the 'return new StatusCodeResult(200) case.
+                if (TryGetParameterStatusCode(semanticModel, creation, creation.ArgumentList, cancellationToken, out statusCode))
+                {
+                    return new ActualApiResponseMetadata(returnStatementSyntax, statusCode);
+                }
+            }
+
             return null;
         }
 
+        private static bool TryGetInitializerStatusCode(
+            SemanticModel semanticModel,
+            InitializerExpressionSyntax initializer,
+            CancellationToken cancellationToken,
+            out int statusCode)
+        {
+            if (initializer == null)
+            {
+                statusCode = default;
+                return false;
+            }
+
+            foreach (var assignment in initializer.Expressions.OfType<AssignmentExpressionSyntax>())
+            {
+                if (assignment.Left is IdentifierNameSyntax identifier)
+                {
+                    var symbolInfo = semanticModel.GetSymbolInfo(identifier, cancellationToken);
+
+                    if (symbolInfo.Symbol is IPropertySymbol property && property.Name.Equals("StatusCode"))
```

Comment:
Ideally this should also check the type, but the cases I checked used a `Nullable<Int32>`, which makes it a bit harder to check. Do we care?